File: day1/main1.py
with open("input") as f:
    ans = 0
    lines = [x for x in f.readlines()]
    for line in lines:
        nums = [x for x in line if x.isdigit()]
        ans += 10*int(nums[0])+int(nums[-1])
    print(ans)

File: day1/main1.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
  ifstream ifs("input");
  string line;
  int ans=0;
  while(getline(ifs, line)){
    ans += 10*(line[line.find_first_of("0123456789")]-'0');
    ans += line[line.find_last_of("0123456789")]-'0';
  }
  cout << ans << endl;
}
